test/shared/com/xilinx/rapidwright/util/VivadoToolsHelper.java

```diff
@@ -68,4 +68,24 @@ public static void assertCanBeFullyPlacedAndRoutedByVivado(Design design, Path d
         ReportRouteStatusResult rrs = VivadoTools.placeAndRouteDesignAndGetStatus(design, dir);
         Assertions.assertTrue(rrs.isFullyRouted());
     }
+
+    public static void assertSamePortCountAfterRoundTripInVivado(Design design, Path dir) {
+        if (!FileTools.isVivadoOnPath()) {
+            return;
+        }
+        int beforePortCount = design.getNetlist().getTopCell().getPorts().size();
+        Design newDesign = VivadoTools.roundTripDCPInVivado(design, dir);
+        int afterPortCount = newDesign.getNetlist().getTopCell().getPorts().size();
+        Assertions.assertEquals(beforePortCount, afterPortCount);
+    }
+
+    public static void assertDifferentPortCountAfterRoundTripInVivado(Design design, Path dir) {
```

Comment:
These two methods are quite similar.  Could we simplify by creating a unified method that returns the number of top ports?  In this case the assertion could be moved out to the test.  Alternatively, you could have another parameter `boolean expectEquals` to simplify the two methods.